src/main/java/com/artipie/rpm/meta/XmlAlter.java

```diff
@@ -47,87 +47,138 @@
+            try (
+                InputStream input = Files.newInputStream(this.file);
+                OutputStream out = Files.newOutputStream(trf)) {
+                new Stream(input, out).pkgAttr(tag, value);
+            }
+            Files.move(trf, this.file, StandardCopyOption.REPLACE_EXISTING);
+        }
+
     }
 
     /**
-     * Updates packages attribute of the given file.
-     * @param tag Tag to change
-     * @param value Value for the attribute
-     * @throws IOException When error occurs
+     * Implementation of {@link XmlAlter} that works with streams, it reads data from
+     * provided InputStream, alters tag attribute and writes result into OutputStream.
+     * @since 1.4
      */
-    public void pkgAttr(final String tag, final String value) throws IOException {
-        final Path trf = Files.createTempFile("", ".xml");
-        try (
-            InputStream input = Files.newInputStream(this.file);
-            OutputStream out = Files.newOutputStream(trf)) {
-            final XMLEventReader reader = XMLInputFactory.newInstance().createXMLEventReader(input);
-            final XMLEventWriter writer = XMLOutputFactory.newInstance().createXMLEventWriter(out);
-            XMLEvent event;
-            while (reader.hasNext()) {
-                event = reader.nextEvent();
-                if (event.isStartElement()
-                    && event.asStartElement().getName().getLocalPart().equals(tag)) {
-                    writer.add(alterEvent(event, value));
-                } else {
-                    writer.add(event);
+    final class Stream implements XmlAlter {
+
+        /**
+         * Input.
+         */
+        private final InputStream input;
+
+        /**
+         * Output.
+         */
+        private final OutputStream out;
+
+        /**
+         * Ctor.
+         * @param input Input to read data from
+         * @param out Where to write the result
+         */
+        public Stream(final InputStream input, final OutputStream out) {
+            this.input = input;
+            this.out = out;
+        }
+
+        @Override
+        public void pkgAttr(final String tag, final String value) {
+            try {
+                final XMLEventReader reader = XMLInputFactory.newInstance()
+                    .createXMLEventReader(this.input);
+                final XMLEventWriter writer = XMLOutputFactory.newInstance()
+                    .createXMLEventWriter(this.out);
+                XMLEvent event;
+                while (reader.hasNext()) {
+                    event = reader.nextEvent();
+                    if (event.isStartElement()
+                        && event.asStartElement().getName().getLocalPart().equals(tag)) {
+                        writer.add(alterEvent(event, value));
+                    } else {
+                        writer.add(event);
+                    }
                 }
+                reader.close();
+                writer.close();
```

Comment:
@olenagerasimova I'd suggest to move `reader.close()` and `writer.close()` into `finally` section of nested `try {...} finally { ... }`